src/sage/groups/perm_gps/partn_ref/data_structures.pxd

```diff
@@ -141,29 +141,31 @@ cdef inline void OP_make_set(OrbitPartition *OP) noexcept:
     cdef int i, n = OP.degree
     cdef int *new_parent, *new_rank, *new_mcr, *new_size
 
-    cdef int *int_array = <int *> sig_malloc( 4*(n+1) * sizeof(int) )
-    if int_array is not NULL:
+    cdef int *int_array = <int *> sig_malloc(4*(n+1) * sizeof(int))
+    if int_array is NULL:
+        raise MemoryError("MemoryError allocating int_array in make_set method")
+    else:
```

Comment:
actually, you can remove the `else` and reduce the indentation of the code. Indeed, the code will be executed only if `int_array` is not `NULL`.

Reply:
That's a good point. I'll switch it over.